csrc/cuda_ops/indices_copy_pad.cu

```diff
@@ -0,0 +1,74 @@
+#include <torch/all.h>
+#include <c10/cuda/CUDAStream.h>
+#include "indices_copy_pad.cuh"
+
+__global__ void copy_and_pad_kernel(
+    const uint16_t* __restrict__ in_hiddens,
+    const int32_t* __restrict__ in_m_indices,
+    uint16_t* __restrict__ out_hiddens,
+    int32_t* __restrict__ out_m_indices,
+    int bs, int bucket_bs, int hidden_size
+) {
+    // 1D grid of blocks handling rows.
+    // blockDim.x = 16 (rows per block)
+    // blockDim.y = 32 (threads per row for parallel copy)
+    
+    int row_in_block = threadIdx.x;
+    int row = blockIdx.x * blockDim.x + row_in_block;
+
+    if (row >= bucket_bs) return;
+
+    bool valid = row < bs;
+
+    // 1. Copy m_indices (One thread per row handles this)
+    if (threadIdx.y == 0) {
+        if (valid) {
+            out_m_indices[row] = in_m_indices[row];
+        } else {
+            out_m_indices[row] = 0;
```

Comment:
Is it correct that padded indices are set to 0?

I remember deepgeem requires that tokens of same indices should be contiguous. It it true?